File: src/main/java/com/smart/appsa/model/enums/CorLamina.java
package com.smart.appsa.model.enums;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum CorLamina {
    VERMELHO(1),
    AZUL(2),
    AMARELO(3),
    VERDE(4),
    PRETO(5),
    BRANCO(6);

    Integer value;

    @JsonCreator
    public static CorLamina fromValue(String value) {
        Integer parsed = Integer.valueOf(value);
        for (CorLamina c : values()) {
            if (parsed.equals(c.value)) return c;
        }
        throw new IllegalArgumentException("CorLamina inválida: " + value);
    }

    private CorLamina(Integer value) {
        this.value = value;
    }

    public Integer getValue() { return value; }
}

File: frontend/planta-smart-4.0/src/domain/enums/CorLamina.ts
export enum CorLamina {
  Vermelho = 'vermelho',
  Azul = 'azul',
  Amarelo = 'amarelo',
  Verde = 'verde',
  Preto = 'preto',
  Branco = 'branco',
}

export const CorLaminaToInt: Record<CorLamina, number> = {
  [CorLamina.Vermelho]: 1,
  [CorLamina.Azul]: 2,
  [CorLamina.Amarelo]: 3,
  [CorLamina.Verde]: 4,
  [CorLamina.Preto]: 5,
  [CorLamina.Branco]: 6,
};

export const CorLaminaStringToEnum: Record<string, CorLamina> = {
  VERMELHO: CorLamina.Vermelho,
  AZUL: CorLamina.Azul,
  AMARELO: CorLamina.Amarelo,
  VERDE: CorLamina.Verde,
  PRETO: CorLamina.Preto,
  BRANCO: CorLamina.Branco,
};
